### PR TITLE
PHP SDK

### PR DESCRIPTION
**Pull request by bencroker**
This is the start of the PHP SDK. Most of it has been fleshed out, following the [SDK spec](https://github.com/starfederation/datastar/blob/main/design/SDK.md), with a few minor things remaining.

### CODE REVIEW

**Review by delaneyj — APPROVED** [review]

File: code/php/sdk/src/ParseIncoming.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+ * @copyright Copyright (c) PutYourLightsOn
+ */
+
+namespace starfederation\datastar;
+
+class ParseIncoming
+{
+    public static function store(): array
+    {
+        return $_GET['datastar'] ?? $_POST['datastar'] ?? [];
```
Comment (Superpat) [inline code/php/sdk/src/ParseIncoming.php:12]: Does $_POST allow parsing of the Put body as well ?

File: code/php/sdk/src/events/Console.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * @copyright Copyright (c) PutYourLightsOn
+ */
+
+namespace starfederation\datastar\events;
+
+use putyourlightson\datastar\events\EventInterface;
```
Comment (Superpat) [inline code/php/sdk/src/events/Console.php:8]: What is putyourlighton  ?

**Review by Superpat — COMMENTED** [review]
My first read through seems good,  I've left a few questions.

File: code/php/sdk/src/ParseIncoming.php (hunk 1 above)
Comment (bencroker) [reply code/php/sdk/src/ParseIncoming.php:12]: No, good catch! Have allowed all methods in https://github.com/starfederation/datastar/commit/c7dcc10f1024efba2059a8db22e52333feb9fdae#diff-361edcffabe63cf52cf35497faa02bc43b7e43e19cd107d82344013e51cf8503

File: code/php/sdk/src/events/Console.php (hunk 2 above)
Comment (bencroker) [reply code/php/sdk/src/events/Console.php:8]: That’s a leftover from the organisation behind https://github.com/putyourlightson/datastar-php. Fixed in https://github.com/starfederation/datastar/commit/c7dcc10f1024efba2059a8db22e52333feb9fdae.

**Pull request merged**